tool/net/redbean.c

```diff
@@ -3720,6 +3725,41 @@ static int LuaFetch(lua_State *L) {
     maxredirects = luaL_optinteger(L, -1, maxredirects);
     lua_getfield(L, 2, "numredirects");
     numredirects = luaL_optinteger(L, -1, numredirects);
+    lua_getfield(L, 2, "headers");
+    if (!lua_isnil(L, -1)) {
+      if (!lua_istable(L, -1))
+        return luaL_argerror(L, 2, "invalid headers value; table expected");
+
+      lua_pushnil(L);
+      while (lua_next(L, -2)) {
+        if (lua_type(L, -2) == LUA_TSTRING) {  // skip any non-string keys
+          key = lua_tolstring(L, -2, &keylen);
+          if (!IsValidHttpToken(key, keylen))
+            return luaL_argerror(L, 2, "invalid header name");
+
+          val = lua_tolstring(L, -1, &vallen);
+          if (!(hdr = gc(EncodeHttpHeaderValue(val, vallen, 0))))
+            return luaL_argerror(L, 2, "invalid header value encoding");
+
+          // Content-Length and Connection will be overwritten;
+          // skip them to avoid duplicates
+          if (strcasecmp(key, "content-length") &&
+              strcasecmp(key, "connection")) {
+            if (!strcasecmp(key, "user-agent")) {
+              agenthdr = gc(strdup(hdr));
```

Comment:
strdup shouldn't be necessary here because gc() is function scoped, rather than block scoped.

Reply:
oh, right; I can simply do `agenthdr = hdr` and `hosthdr = hdr`, since `hdr` values will be `gc` collected anyway.